### DISCUSSION TITLE
Allow for Type Unions for Lists (RuntimeError: Type not yet supported: int | str)

### DISCUSSION THREAD

**Discussion by fordmcdonald** [fastapi/typer#655] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
skipseq: List[int | str] = typer.Option(
        [],
        "-s",
        "--skipseq",
        help="Exclude this sequence, can be specified multiple times",
    )
```

### Description

When outlining parameters of type List, they should support the type unions allowed by Python.  A list can contain multiple types; therefore, Typer should allow for this behavior, as well.

### Operating System

macOS

### Typer Version

0.6.1

### Python Version

Python 3.10.12

**Comment by ambroisie** [comment]
It's especially troubling for `<type> | None`, which is not supported when `Optional[<type>]` is.

**Reply by ambroisie** [reply]
This is tracked in #533.
